Components/FlightControl/Inc/RocketSM.hpp

```diff
@@ -0,0 +1,295 @@
+    RS_FILL,        // N2 Prefill/Purge/Leak-check/Load-cell Tare check sub-sequences, full control of valves (except MEV) allowed
+    RS_ARM,         // We don't allow fill etc. 1-2 minutes before launch : Cannot fill rocket with N2 etc. unless you return to FILL
+                    // Power Transition, Fill-Arm Disconnect Sub-sequences (you should be able to revert the power transition)
+
+    //-- IGNITION -- Manual venting NOT ALLOWED
+    RS_IGNITION,    // Ignition of the ignitors
+    RS_LAUNCH,      // Launch triggered by confirmation of ignition (from ignitor) is nominal : MEV Open Sequence
+
+    //-- BURN --
+    // Vents should stay closed, manual venting NOT ALLOWED
+    // !vents open is definitely not ideal for abort! Best to keep it closed with manual override if we fail here
+    // (can we maybe have the code change the true default state by overwriting EEPROM?)
+    // Ideally we don't want to EVER exceed 7 seconds of burn time (we should store this time at the very least - split into 1 sub-stage for each second if necessary).
+    // For timing we want ~1/10th of a second or better.
+    RS_BURN,        // Main burn (vents closed MEV open) - 5-6 seconds (TBD) :
+                    
+
+    //-- COAST --
+    // Manual venting NOT ALLOWED
+    RS_COAST,       // Coasting (MEV closed, vents closed) - 30 seconds (TBD) ^ Vents closed applies here too, in part. Includes APOGEE
+
+    //-- DESCENT / POSTAPOGEE --
+    // Automatic Venting AND Vent Control ALLOWED
+    RS_DESCENT,  // Vents open (well into the descent)
+    RS_RECOVERY,  // Vents open, MEV closed, transmit all data over radio and accept vent commands
+
+    //-- RECOVERY / TECHNICAL --
+    RS_ABORT,       // Abort sequence, vents open, MEV closed, ignitors off
+    RS_NONE         // Invalid state, must be last
+};
+
+/**
+ * @brief Rocket State Commands, all fall under the GLOBAL_COMMAND -> STATE_ACTION umbrella
+ *
+ *        All commands in this list are only activated if the state allows them.
+ */
+enum RocketStateCommands
+{
+    //-- STATE TRANSITION --
+    RST_ABORT,      // Abort the rocket - transition to ABORT if available
+
+
+    RSC_MAX         // Invalid command, must be last
+};
+
+/**
+ * @brief External Rocket Control Commands, all fall under GLOBAL_COMMAND -> CONTROL_ACTION umbrella
+ *
+ *        State specific commands, must be all in-order to avoid duplicate command IDs
+ */
+enum RocketControlCommands
+{
+    //-- PRE-IGNITION and RECOVERY --
+    RSC_ABORT,       // Transition to ABORT state - available from all states except for IGNITION/LAUNCH/BURN
+    RSC_OPEN_VENT,   // Open the vent valve
+    RSC_CLOSE_VENT,  // Close the vent valve
+    RSC_OPEN_DRAIN,  // Open the drain valve
+    RSC_CLOSE_DRAIN, // Close the drain valve
+    RSC_MEV_CLOSE,   // Forces MEV to close - ONLY supported in states where it is safe to close the MEV
+
+    //-- PRELAUNCH --
+    RSC_BEGIN_FILL, // Transition to the FILL state
```

Comment:
let's follow a naming convention for the Transition-To-State commands (ie. it could literally just be RSC_TRANSITION_TO_FILL)

Reply:
Added TRANSITION to the name of state change commands: 1ad467b9575ad0deb3796a3431f839adda189b27

Note that I kept a little more context in the names eg. "CONFIRM_IGNITION" instead of simply "IGNITION" for state transitions that imply a kind of process